### ISSUE TITLE
Mac app preventing sleep

### ISSUE THREAD

**Issue by mnium** [home-assistant/iOS#1297]
Model Name: MacBook Pro (Retina, 15-inch, Late 2013)
macOS Version: 10.15.7
App Version: 2020.7(10)
Safari Version: 14.0.1
Home Assistant Core Version: 0.118.5

Bug: When starting the mac app three processes are opened. Home Assistant, com.apple.WebKit.Networking and com.apple.WebKit.WebContent. The Webkit processes are preventing the mac from sleeping as long as the Home Assistant app is open.

Additional: Blocked connections to googleapis.com and sentry.io with Little Snitch 

Expected behavior: Let mac go to sleep when app is open

Screenshot:
<img width="374" alt="Screenshot" src="https://user-images.githubusercontent.com/73353833/101664804-2e113980-3a44-11eb-9698-4b7555dfc847.png">

**Comment by jazzmonger** [comment]
ok, help me out here - where is the link to DL the MacOS app???

**Comment by SeanPM5** [comment]
@jazzmonger you can find it on this repo by clicking the "Code" tab up top and then in the right sidebar part there's a section called Releases. Or you can just click here: https://github.com/home-assistant/iOS/releases

Once you install the macOS app, it has an update function built-in that will alert you of new releases.

**Comment by jazzmonger** [comment]
Perfect - works great so far..  where does it show what version is running?  no App config button...

**Comment by jazzmonger** [comment]
never mind...  Mac /about.  :). just like it should be.

**Comment by zacwest** [comment]
Looking at the reasons (using `pmset -g assertions`), it appears that the webview here is preventing sleep due to:

```
   pid 176(runningboardd): [0x00041cf50001a6b2] 01:06:56 PreventUserIdleSystemSleep named: "xpcservice<com.apple.WebKit.Networking([application<io.robbie.HomeAssistant.dev(501)>:24126])(501)>:24131176-24126-110811:NetworkProcess Foreground Assertion"
	Created for PID: 24131.
   pid 176(runningboardd): [0x00041cf50001a6b1] 01:06:56 PreventUserIdleSystemSleep named: "xpcservice<com.apple.WebKit.WebContent([application<io.robbie.HomeAssistant.dev(501)>:24126])(501)>:24130176-24126-110808:WebProcess Foreground Assertion"
	Created for PID: 24130.
```

I believe this is an Apple bug tracing back to the iOS origins of the Catalyst, as evidenced by the fact that 'Hide'-ing the app (which triggers the 'background' mode for iOS apps on macOS) causes these assertions to go away and allows the app to sleep.

I've filed this with Apple as FB8958507. I don't believe we're going to be able to work around this since it's out of our process. I'll attach any further correspondence with Apple here.

**Comment by mnium** [comment]
Thanks!
Additional Information: When the app is minimized, it no longer is preventing sleep. This might confirm your theory with the bug in Catalyst.

**Comment by jcam** [comment]
Any word back on this? I realized this is what was causing my dead batteries over the last couple of weeks

**Comment by zacwest** [comment]
Apple indicates this may be fixed in macOS 12. I only have the 1 Mac so I haven't upgraded yet, but if you have, you can run the `pmset -g assertions` command to see if the WebView is still requesting the system not sleep.

**Comment by zacwest** [comment]
I can confirm this is resolved in macOS 12.0.
